src/pages/api/iexec/new-proposal.tsx

```diff
@@ -0,0 +1,104 @@
+import mongoose from 'mongoose';
+import { getProposalsFromPlatformServices } from '../../../queries/proposals';
+import { EmailType, IProposal } from '../../../types';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { NewProposalEmail } from '../../../modules/Web3Mail/schemas/new-proposal-model';
+import { sendMailToAddresses } from '../../../scripts/iexec/sendMailToAddresses';
+import { Timestamp } from '../../../modules/Web3Mail/schemas/timestamp-model';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const mongoUri = process.env.NEXT_MONGO_URI;
+  const TIMESTAMP_NOW_SECONDS = Math.floor(new Date().getTime() / 1000);
+
+  // Check whether the key is valid
+  const key = req.query.key;
+  if (key !== process.env.NEXT_PRIVATE_CRON_KEY) {
+    return res.status(401).json({ message: 'Unauthorized' });
+  }
+
+  if (!mongoUri) {
+    throw new Error('MongoDb URI is not set');
+  }
+  await mongoose.connect(mongoUri as string);
+
+  const platformId = process.env.NEXT_PUBLIC_PLATFORM_ID;
+
+  if (!platformId) {
+    throw new Error('Platform Id is not set');
+  }
+
+  try {
+    //Get latest timestamp from DB if exists
+    let timestamp = await Timestamp.findOne({ type: EmailType.NewProposal });
+    if (!timestamp) {
+      timestamp = await Timestamp.create({
+        type: EmailType.NewProposal,
+        date: `${TIMESTAMP_NOW_SECONDS - 3600 * 24}`,
+      });
+      timestamp.save();
+    }
+    const timestampValue = timestamp.date;
+
+    // Overrite timestamp with new value
+    await Timestamp.updateOne(
+      { type: EmailType.NewProposal },
+      { date: `${TIMESTAMP_NOW_SECONDS}` },
+    );
+    //TODO Uncomment
+    const response = await getProposalsFromPlatformServices(platformId, '0');
+    // const response = await getProposalsFromPlatformServices(platformId, timestampValue);
```

Comment:
- get for the newProposal type the cron schedule form vercel.json
- use it to define the date filter here, and a a multiplicator to handle retry in case some thing bad happened (add comment in the code to explain that) 
- ex: cron is 1 hour, we have a filter date to get all new proposal since  5 hours, like that if any errors happened we will have 5 retries

Reply:
the user can also decide to not have a vercel.json and add the sinceTimestamp param: /api/iexec/new-proposal?key=racoonKey&sinceTimestamp=12345454

Reply:
if no vercel.json and no param in query, return an error